homeassistant/components/insteon/api/aldb.py

```diff
@@ -0,0 +1,277 @@
+    """Create an All-Link Database record for an Insteon device."""
+    device = devices[msg[DEVICE_ADDRESS]]
+    if not device:
+        notify_device_not_found(connection, msg, INSTEON_DEVICE_NOT_FOUND)
+        return
+
+    record = msg[ALDB_RECORD]
+    device.aldb.add(
+        group=record["group"],
+        controller=record["is_controller"],
+        target=record["target"],
+        data1=record["data1"],
+        data2=record["data2"],
+        data3=record["data3"],
+    )
+    connection.send_result(msg[ID])
+
+
+@websocket_api.require_admin
+@websocket_api.async_response
+@websocket_api.websocket_command(
+    {
+        vol.Required(TYPE): "insteon/aldb/write",
+        vol.Required(DEVICE_ADDRESS): str,
+    }
+)
+async def websocket_write_aldb(hass, connection, msg):
+    """Create an All-Link Database record for an Insteon device."""
+    device = devices[msg[DEVICE_ADDRESS]]
+    if not device:
+        notify_device_not_found(connection, msg, INSTEON_DEVICE_NOT_FOUND)
+        return
+
+    await device.aldb.async_write()
+    hass.async_create_task(async_reload_and_save_aldb(hass, device))
+    connection.send_result(msg[ID])
+
+
+@websocket_api.require_admin
+@websocket_api.async_response
+@websocket_api.websocket_command(
+    {
+        vol.Required(TYPE): "insteon/aldb/load",
+        vol.Required(DEVICE_ADDRESS): str,
+    }
+)
+async def websocket_load_aldb(hass, connection, msg):
+    """Create an All-Link Database record for an Insteon device."""
+    device = devices[msg[DEVICE_ADDRESS]]
+    if not device:
+        notify_device_not_found(connection, msg, INSTEON_DEVICE_NOT_FOUND)
+        return
+
+    hass.async_create_task(async_reload_and_save_aldb(hass, device))
+    connection.send_result(msg[ID])
+
+
+@websocket_api.require_admin
+@websocket_api.async_response
+@websocket_api.websocket_command(
+    {
+        vol.Required(TYPE): "insteon/aldb/reset",
+        vol.Required(DEVICE_ADDRESS): str,
+    }
+)
+async def websocket_reset_aldb(hass, connection, msg):
+    """Create an All-Link Database record for an Insteon device."""
+    device = devices[msg[DEVICE_ADDRESS]]
+    if not device:
+        notify_device_not_found(connection, msg, INSTEON_DEVICE_NOT_FOUND)
+        return
+
+    device.aldb.clear_pending()
+    connection.send_result(msg[ID])
+
+
+@websocket_api.require_admin
+@websocket_api.async_response
+@websocket_api.websocket_command(
+    {
+        vol.Required(TYPE): "insteon/aldb/add_default_links",
+        vol.Required(DEVICE_ADDRESS): str,
+    }
+)
+async def websocket_add_default_links(hass, connection, msg):
+    """Add the default All-Link Database records for an Insteon device."""
+    device = devices[msg[DEVICE_ADDRESS]]
+    if not device:
+        notify_device_not_found(connection, msg, INSTEON_DEVICE_NOT_FOUND)
+        return
+
+    device.aldb.clear_pending()
+    await device.async_add_default_links()
+    hass.async_create_task(async_reload_and_save_aldb(hass, device))
```

Comment:
Should we give the user feedback if this fails instead of doing it in the background?